src/Tools/dotnet-gcdump/CommandLine/CollectCommandHandler.cs

```diff
@@ -24,24 +24,40 @@ internal static class CollectCommandHandler
         /// <param name="processId">The process to collect the gcdump from.</param>
         /// <param name="output">The output path for the collected gcdump.</param>
         /// <returns></returns>
-        private static async Task<int> Collect(CancellationToken ct, IConsole console, int processId, string output, int timeout, bool verbose)
+        private static async Task<int> Collect(CancellationToken ct, IConsole console, int processId, string transportPath, string output, int timeout, bool verbose)
         {
             try
             {
-                if (processId < 0)
+                if (string.IsNullOrEmpty(transportPath))
                 {
-                    Console.Out.WriteLine($"The PID cannot be negative: {processId}");
-                    return -1;
+                    if (processId < 0)
+                    {
+                        Console.Out.WriteLine($"The PID cannot be negative: {processId}");
+                        return -1;
+                    }
+
+                    if (processId == 0)
+                    {
+                        Console.Out.WriteLine($"-p|--process-id is required");
+                        return -1;
+                    }
                 }
-
-                if (processId == 0)
+                else
                 {
-                    Console.Out.WriteLine($"-p|--process-id is required");
-                    return -1;
+                    if (!File.Exists(transportPath) && !File.Exists(@"\\.\pipe\" + transportPath))
+                    {
+                        Console.Error.WriteLine("Requested transport does not exist");
+                        return -1;
+                    }
+                    else if (processId != 0)
+                    {
+                        Console.Error.WriteLine("Cannot specify both a PID and a specific transport");
+                        return -1;
+                    }
                 }
 
                 output = string.IsNullOrEmpty(output) ? 
-                    $"{DateTime.Now.ToString(@"yyyyMMdd\_hhmmss")}_{processId}.gcdump" :
+                    $"{DateTime.Now.ToString(@"yyyyMMdd\_hhmmss")}_{(processId != 0 ? processId.ToString() : (new FileInfo(transportPath)).Name)}.gcdump" :
```

Comment:
I'd guess that typical port names will always be the same and thus won't be useful to distinguish processes. What do you think of {timestamp}.gcdump as the filename when a transport path is used?